src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -12,7 +12,7 @@ use std::collections::{BTreeMap, BTreeSet};
 use std::iter;
 use std::num::{NonZeroI64, NonZeroUsize};
 use std::panic::AssertUnwindSafe;
-use std::sync::Arc;
+use std::sync::{Arc, OnceLock};
```

Comment:
@mjibson isn't `OnceLock` a good mechanism to bring the extra `CatalogState` maps that keep around optimized and physical plans back under the `CatalogState` item structs?

Reply:
Maybe, but mostly I think that's not better than what we have today. I like that `CatalogState` feels like a POCO snapshot object that fully represents the state of the catalog at a specific transient revision. Adding OnceLock to it removes the invariant that it only ever mutates when the transient revision increases.